### PR TITLE
[uptraceexporter] Updated for upcoming client auth extension changes

### PR DESCRIPTION
**Pull request by gramidt**
**Description:** 
Updated uptraceexporter for upcoming client auth extension changes

**Link to tracking Issue:**
#3507

**Testing:**
Unit tests

**Documentation:**
Unit tests

### CODE REVIEW

File: exporter/uptraceexporter/exporter.go [hunk 1]
```diff
@@ -119,6 +99,30 @@ func (e *traceExporter) pushTraceData(ctx context.Context, traces pdata.Traces)
 	return nil
 }
 
+func (e *traceExporter) Start(_ context.Context, _ component.Host) error {
```
Comment (bogdandrutu) [inline exporter/uptraceexporter/exporter.go:102]: This can be hidden `start` correct?

File: exporter/uptraceexporter/factory.go [hunk 2]
```diff
@@ -74,5 +71,6 @@ func createTracesExporter(
 		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
 		exporterhelper.WithRetry(oCfg.RetrySettings),
 		exporterhelper.WithQueue(oCfg.QueueSettings),
+		exporterhelper.WithStart(exporter.Start),
 		exporterhelper.WithShutdown(exporter.Shutdown))
```
Comment (bogdandrutu) [inline exporter/uptraceexporter/factory.go:75]: same the Shutdown can be `shutdown`

File: exporter/uptraceexporter/exporter_test.go [hunk 3]
```diff
@@ -35,17 +36,21 @@ import (
...
-	exp, err := newTracesExporter(cfg, zap.NewNop())
+	exp := newTracesExporter(cfg, zap.NewNop())
+	require.NotNil(t, exp)
+
+	err := exp.Start(context.Background(), componenttest.NewNopHost())
 	require.Error(t, err)
-	require.Nil(t, exp)
 }
 
 func TestNewTracesExporterEndpoint(t *testing.T) {
```
Comment (bogdandrutu) [inline exporter/uptraceexporter/exporter_test.go:46]: Not clear the testing case compare with previous test. Probably this needs to be named invalid endpoint and previous test NoEndpoint?

File: exporter/uptraceexporter/exporter.go (hunk 1 above)
Comment (gramidt) [reply exporter/uptraceexporter/exporter.go:102]: I do not see a reason it shouldn't be. I just kept the pattern that was implemented with their "Shutdown"

File: exporter/uptraceexporter/factory.go (hunk 2 above)
Comment (gramidt) [reply exporter/uptraceexporter/factory.go:75]: Agreed. I did not want to change their existing implementation when doing this PR, just wanted the minimal changes needed. I will happy change `Shutdown` too.

File: exporter/uptraceexporter/exporter.go [hunk 4]
```diff
@@ -119,7 +99,31 @@ func (e *traceExporter) pushTraceData(ctx context.Context, traces pdata.Traces)
 	return nil
 }
 
-func (e *traceExporter) Shutdown(ctx context.Context) error {
+func (e *traceExporter) start(_ context.Context, _ component.Host) error {
+	if e.cfg.HTTPClientSettings.Endpoint != "" {
+		e.logger.Warn("uptraceexporter: endpoint is not supported; use dsn instead")
+	}
```
Comment (bogdandrutu) [inline exporter/uptraceexporter/exporter.go:105]: This should be part of the validation, probably a separate PR.

**Review by bogdandrutu — APPROVED** [review]

File: exporter/uptraceexporter/exporter.go (hunk 4 above)
Comment (gramidt) [reply exporter/uptraceexporter/exporter.go:105]: Agreed. I just moved the original logic.

**Pull request merged**